be/src/exec/workgroup/work_group.cpp

```diff
@@ -111,20 +118,81 @@ void WorkGroupManager::destroy() {
+
+    if (_wg_metrics.count(wg->name()) == 0) {
+        //cpu limit
+        auto resource_group_cpu_limit_ratio = std::make_unique<starrocks::DoubleGauge>(MetricUnit::PERCENT);
+        //cpu concurrent
+        auto resource_group_cpu_use_ratio = std::make_unique<starrocks::DoubleGauge>(MetricUnit::PERCENT);
+        //mem limit
+        auto resource_group_mem_limit_bytes = std::make_unique<starrocks::IntGauge>(MetricUnit::BYTES);
+        //mem concurrent
+        auto resource_group_mem_allocated_bytes = std::make_unique<starrocks::IntGauge>(MetricUnit::BYTES);
+
+        StarRocksMetrics::instance()->metrics()->register_metric("resource_group_cpu_limit_ratio",
+                                                                 MetricLabels().add("name", wg->name()),
+                                                                 resource_group_cpu_limit_ratio.get());
+        StarRocksMetrics::instance()->metrics()->register_metric("resource_group_cpu_use_ratio",
+                                                                 MetricLabels().add("name", wg->name()),
+                                                                 resource_group_cpu_use_ratio.get());
+        StarRocksMetrics::instance()->metrics()->register_metric("resource_group_mem_limit_bytes",
+                                                                 MetricLabels().add("name", wg->name()),
+                                                                 resource_group_mem_limit_bytes.get());
+        StarRocksMetrics::instance()->metrics()->register_metric("resource_group_mem_allocated_bytes",
+                                                                 MetricLabels().add("name", wg->name()),
+                                                                 resource_group_mem_allocated_bytes.get());
+
+        _wg_cpu_limit_metrics.emplace(wg->name(), std::move(resource_group_cpu_limit_ratio));
+        _wg_cpu_metrics.emplace(wg->name(), std::move(resource_group_cpu_use_ratio));
+        _wg_mem_limit_metrics.emplace(wg->name(), std::move(resource_group_mem_limit_bytes));
+        _wg_mem_metrics.emplace(wg->name(), std::move(resource_group_mem_allocated_bytes));
+    }
+    _wg_metrics.emplace(wg->name(), wg->unique_id());
+}
+
+void WorkGroupManager::update_metrics_unlocked() {
+    for (auto& wg_metric : _wg_metrics) {
+        auto wg = _workgroups.find(wg_metric.second);
+        if (wg != _workgroups.end()) {
+            _wg_cpu_limit_metrics.find(wg_metric.first)->second->set_value(wg->second->get_cpu_expected_use_ratio());
+            _wg_cpu_metrics.find(wg_metric.first)->second->set_value(wg->second->get_cpu_actual_use_ratio());
+            _wg_mem_limit_metrics.find(wg_metric.first)->second->set_value(wg->second->mem_limit());
+            _wg_mem_metrics.find(wg_metric.first)->second->set_value(wg->second->mem_tracker()->consumption());
+        } else {
+            _wg_cpu_limit_metrics.find(wg_metric.first)->second->set_value(0);
```

Comment:
why need to set 0 value explicitly?

Reply:
Because if the workgroup is not found in the _workgroups map, it means that the workgroup has been removed, but I considered that the workgroup with the same name may be rejoined later,in order to keep the collected metrics data continuous, so I chose to reset the metrics value to 0 instead of  delete .